### PR TITLE
Misc: Modify deploy script to match with gas less blockchain

### PR DESCRIPTION
**Pull request by dqtkien**

### CODE REVIEW

File: tasks/deploy-orochi-network.ts [hunk 1]
```diff
@@ -6,17 +6,33 @@ import { env } from '../env';
...
+    }
+    const needCustomProvider = CHAIN_NEED_CUSTOM_PROVIDER.includes(hre.network.config.chainId);
+    const isGasLessBlockchain = CHAIN_NEED_CUSTOM_PROVIDER.includes(hre.network.config.chainId);
     // Public key
     let pk = env.OROCHI_PUBLIC_KEY.replace(/^0x/gi, '').trim();
     let correspondingAddress = getAddress(`0x${keccak256(`0x${pk.substring(2, 130)}`).substring(26, 66)}`);
     // Get deployer account
-    const { chainId } = await hre.ethers.provider.getNetwork();
-    const account = await getWallet(hre, chainId);
+    const provider = needCustomProvider
```
Comment (chiro-hiro) [inline tasks/deploy-orochi-network.ts]: This code looks so ugly, we can move the logic into `getWallet()` method, it will be better and clean.

**Comment by chiro-hiro** [comment]
https://github.com/orochi-network/smart-contracts/pull/40/files#diff-55c5cf1d7cec530bbc2f0baa66a3cd3ff3b58a0483ec63d1c1515c2b110d5705R76-R82

This code looks so horrible, agree that we need thing to be done quick but it shouldn't be done in this way.

File: tasks/deploy-orochi-network.ts (hunk 1 above)
Comment (dqtkien) [reply tasks/deploy-orochi-network.ts]: Fixed

File: helpers/provider.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      };
+    }
+
+    const result: AxiosResponse<JsonRpcResult> = await axios.request({
+      method: 'POST',
+      headers,
+      url: this.url,
+      data: payload,
+    });
+    console.log(`Request to: ${this.url} payload:`, payload, result.data);
```
Comment (chiro-hiro) [inline helpers/provider.ts]: @dqtkien Please remove debug

Comment (dqtkien) [reply helpers/provider.ts]: Fixed

**Review by chiro-hiro — APPROVED** [review]

**Pull request merged**